homeassistant/components/zwave_me/__init__.py

```diff
@@ -0,0 +1,122 @@
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+class ZWaveMeController:
+    """Main ZWave-Me API class."""
+
+    def __init__(self, hass: HomeAssistant, config: ConfigEntry) -> None:
+        """Create the API instance."""
+        self.device_ids: list[str] = []
+        self._hass = hass
+        self._config = config
+        self.zwave_api = ZWaveMe(
+            on_device_create=self.on_device_create,
+            on_device_update=self.on_device_update,
+            on_new_device=self.add_device,
+            token=self._config.data["token"],
+            url=self._config.data["url"],
+            platforms=ZWAVEPLATFORMS,
+        )
+
+    def add_device(self, device: ZWaveMeData):
+        """Send signal to create device."""
+        if device.deviceType in ZWAVEPLATFORMS:
+            if device.id in self.device_ids:
+                dispatcher_send(self._hass, "ZWAVE_ME_INFO_" + device.id, device)
+            else:
+                dispatcher_send(
+                    self._hass, "ZWAVE_ME_NEW_" + device.deviceType.upper(), device
+                )
+                self.device_ids.append(device.id)
+
+    def on_device_create(self, devices: list[ZWaveMeData]):
+        """Create multiple devices."""
+        if devices is None:
+            return
+        for device in devices:
+            self.add_device(device)
+
+    def on_device_update(self, new_info: ZWaveMeData):
+        """Send signal to update device."""
+        dispatcher_send(self._hass, "ZWAVE_ME_INFO_" + new_info.id, new_info)
+
+
+class ZWaveMeDevice(Entity):
+    """Representation of a ZWaveMe device."""
+
+    def __init__(self, device):
+        """Initialize the device."""
+        self._name = device.title
+        self._outlet = None
+        self.device = device
+
+    async def async_added_to_hass(self) -> None:
+        """Connect to an updater."""
+        async_dispatcher_connect(
+            self.hass, "ZWAVE_ME_INFO_" + self.device.id, self.get_new_data
+        )
+
+    def get_device(self):
+        """Get device info by id."""
+        for device in self.hass.data[DOMAIN].get_devices():
+            if device.id == self.device.id:
+                return device
+
+        return None
+
+    def get_new_data(self, new_data):
+        """Update info in the HAss."""
+        self.device = new_data
+        self.schedule_update_ha_state()
+
+    def get_available(self):
+        """Get availability of a generic device."""
+        return not self.device.isFailed
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return true if device is online."""
+        return self.get_available()
+
+    def update(self):
+        """Update device state."""
+
+    @property
+    def unique_id(self) -> str:
```

Comment:
Assign unique ID in the constructor using `self._attr_unique_id = self.device.id`.

Uniqueness is checked within the scope of your integration + platform so don't include `DOMAIN`.

`name` is a user generated value and is not allowed as part of the name. 

If a device provides 2 sensors, using just the device ID will create non-unique IDs.